server/src/test/java/io/crate/planner/optimizer/rule/EliminateCrossJoinTest.java

```diff
@@ -375,4 +377,21 @@ public void test_do_not_reorder_without_a_crossjoin() throws Exception {
         assertThat(result).isNull();
     }
 
+    @Test
+    public void test_build_graph_fron_ouble_aliased_relation() {
+        var x =  """
+                SELECt * FROM
+                    (SELECT i.indexrelid,
+                           i.indrelid,
+                           i.indisprimary,
+                           information_schema._pg_expandarray(i.indkey) AS keys
+                    FROM   pg_catalog.pg_index i) i
+                    JOIN   pg_catalog.pg_attribute a ON a.attnum = (i.keys).x
+                """;
+
+        LogicalPlan logicalPlan = e.logicalPlan(x);
+        var graph = JoinGraph.create(logicalPlan, Function.identity());
+        assertThat(graph).isNotNull();
```

Comment:
This assertion works also without the fix (`putIfAbsent`), can we please assert more details to make sure the test covers the fix?